python/cudf/cudf/core/dataframe.py

```diff
@@ -2046,30 +2046,25 @@ def _make_operands_and_index_for_binop(
             rhs = {name: other for name in self._data}
             equal_columns = True
         elif isinstance(other, Series):
+            self_pd_columns = self._data.to_pandas_index()
+            other_pd_index = other.index.to_pandas()
             if (
                 not can_reindex
                 and fn in cudf.utils.utils._EQUALITY_OPS
-                and (
-                    not self._data.to_pandas_index().equals(
-                        other.index.to_pandas()
-                    )
-                )
+                and not self_pd_columns.equals(other_pd_index)
```

Comment:
```suggestion
                and not (self_pd_columns := self._data.to_pandas_index()).equals(
                       other_pd_index := other.index.to_pandas()
                )
```

We could walrus this.

Reply:
Good catch. Changed to this